ecosystem/indexer-grpc/indexer-grpc-table-info/src/table_info_service.rs

```diff
@@ -294,6 +337,65 @@ impl TableInfoService {
         Ok(())
     }
 
+    /// Tries to upload a snapshot of the database if the backup service is enabled.
+    /// 1. If current epoch is backuped, it will skip the backup.
+    /// 2. If the chain id in the backup metadata does not match with the current network, it will panic.
+    /// TODO(larry): improve the error handling.
+    async fn try_backup_db_snapshot(
+        context: Arc<Context>,
+        last_version: u64,
+        indexer_async_v2: Arc<IndexerAsyncV2>,
+        backup_restore_operator: Arc<GcsBackupRestoreOperator>,
+    ) {
+        // Validation.
+        let metadata_epoch = backup_restore_operator.clone().get_metadata_epoch();
+        let (_, _, block_event) = context
+            .db
+            .get_block_info_by_version(last_version)
+            .unwrap_or_else(|_| {
+                panic!("Could not get block_info for last version {}", last_version,)
+            });
+        let block_event_epoch = block_event.epoch();
+        if metadata_epoch >= block_event_epoch {
+            // No need to backup if the latest version is already backed up.
```

Comment:
nit: s/version/epoch

still feel a little bit weird here.. since you are basically making the backup at a random version close to the beginning of an epoch

Reply:
yep; the other approach is(within this function):

1. when transactions for new epoch arrive, stop the processing and snapshot the db first.
2. once snapshot is done continue processing table info.
3. backup thread only monitors if a new snapshot exists and if there is, backup.

Pro:
1. a clean cut of snapshot for each epoch
con:
1. table info processing may halt for db snapshotting

The questions are:
1. I have concern about the snapshotting when db size grows. If this is not a big issue, we can try this route.
2. On the other hand, how critical is it to get a clean cut here? i feel it's not a big issue..

Reply:
resolved offline.